assets/src/block-editor/components/sidebar-notification/index.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import './style.css';
+import { Loading } from '../../../components/loading';
+
+/**
+ * Notification component used in the block editor sidebar.
+ *
+ * @param {Object} props
+ * @param {Object} props.action Call to action element.
+ * @param {Object} props.icon Status icon element.
+ * @param {boolean} props.isLoading Flag indicating if it's a loading message.
+ * @param {string} props.message Message text.
+ */
+export function SidebarNotification( {
+	action,
+	icon,
```

Comment:
As stated in the [JSX docs](https://reactjs.org/docs/jsx-in-depth.html#user-defined-components-must-be-capitalized), user-defined components should be capitalized.


```suggestion
	Action,
	Icon,
```

Reply:
I think this applies only to the component definition, not necessarily to the case when a component is passed via props. If such a prop would be capitalized, then it would also have to be capitalized when using a component and it doesn't seem right, i.e.

```js
<SidebarNotification Action={ ... } Icon={ ... } />
```

The official [docs](https://reactjs.org/docs/composition-vs-inheritance.html#containment) don't suggest using capitalized props in the `SplitPane` example:

```js
function SplitPane(props) {
  return (
    <div className="SplitPane">
      <div className="SplitPane-left">
        {props.left}
      </div>
      <div className="SplitPane-right">
        {props.right}
      </div>
    </div>
  );
}

function App() {
  return (
    <SplitPane
      left={
        <Contacts />
      }
      right={
        <Chat />
      } />
  );
}
```

In Gutenberg, such props are starting with a lowercase letter, too (check out an [example](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/placeholder/index.js#L25-L26)).

Reply:
Ah right that makes sense. Thanks for the examples.